syncopy/shared/input_validators.py

```diff
@@ -0,0 +1,273 @@
+# -*- coding: utf-8 -*-
+#
+# Validators for user submitted frontend arguments like foi, taper, etc.
+# Input args are the parameters to check for validity + auxiliary parameters
+# needed for the checks.
+#
+
+# Builtin/3rd party package imports
+import numpy as np
+
+from syncopy.shared.errors import SPYValueError, SPYWarning, SPYInfo
+from syncopy.shared.parsers import data_parser, scalar_parser, array_parser
+from syncopy.shared.const_def import availableTapers, generalParameters
+
+
+def validate_foi(foi, foilim, samplerate):
+
+    '''
+
+    Parameters to check
+    -------------------    
```

Comment:
I guess I could use this:

>8. Other Parameters
>
>An optional section used to describe infrequently used parameters. It should only be used if a function has a large >number of keyword parameters, to prevent cluttering the Parameters section.

to distinguish between parameters which are checked, and the auxiliary ones (needed for the checks but not validated themselves)?!

Reply:
Yes, perfect!

Reply:
Should be all good with 0decee4